thunder/torch/__init__.py

```diff
@@ -1589,6 +1589,56 @@ def where(
     return clang.where(pred, a, b)
 
 
+@torchsymbol(torch.nan_to_num, is_method=True)
+def nan_to_num(
+    a: TensorLike,
+    nan: None | Number = 0.0,
+    posinf: None | Number = None,
+    neginf: None | Number = None,
+    /,
+    out: None | TensorLike = None,
+):
+
+    utils.check(out is None, lambda: "out is not None which is currently unsupported", NotImplementedError)
+
+    if dtypes.is_boolean_dtype(a.dtype):
+        # NOTE PyTorch returns a.clone()
+        return a - 0
+
+    if dtypes.is_integer_dtype(a.dtype):
+        # NOTE PyTorch returns a.clone()
+        return a | a
```

Comment:
Shouldn't it be the other way around? Also, it seems like `a | a` works even with integer types in PyTorch, not sure about Thunder though...

Reply:
Oh wow, my bad. I was blind for a second. Let me fix that in like 2 hours.